pygments/lexers/elpi.py

```diff
@@ -0,0 +1,143 @@
+
+    :copyright: Copyright 2006-2021 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from pygments.lexer import RegexLexer, bygroups, include
+from pygments.token import \
+    Text, Comment, Operator, Keyword, Name, String, Number
+
+__all__ = ['ElpiLexer']
+
+class ElpiLexer(RegexLexer):
+    """
+    Lexer for the `Elpi <http://github.com/LPCIC/elpi>`_ programming language.
+
+    .. versionadded::2.11
+    """
+
+    name = 'Elpi'
+    aliases = ['elpi']
+    filenames = ['*.elpi']
+    mimetypes = ['text/x-elpi']
+    
+    lcase_re = r"[a-z]"
+    ucase_re = r"[A-Z]"
+    digit_re = r"[0-9]"
+    schar2_re = r"(\+|\*|/|\^|<|>|`|'|\?|@|#|~|=|&|!)"
+    schar_re = r"({}|-|\$|_)".format(schar2_re)
+    idchar_re = r"({}|{}|{}|{})".format(lcase_re,ucase_re,digit_re,schar_re)
+    idcharstarns_re = r"({}+|(?=\.[a-z])\.{}+)".format(idchar_re,idchar_re)
+    symbchar_re = r"({}|{}|{}|{}|:)".format(lcase_re, ucase_re, digit_re, schar_re)
+    constant_re = r"({}{}*|{}{}*|{}{}*|_{}+)".format(ucase_re, idchar_re, lcase_re, idcharstarns_re,schar2_re, symbchar_re,idchar_re)
+    symbol_re=r"(,|<=>|->|:-|;|\?-|->|&|=>|\bas\b|\buvar\b|<|=<|=|==|>=|>|\bi<|\bi=<|\bi>=|\bi>|\bis\b|\br<|\br=<|\br>=|\br>|\bs<|\bs=<|\bs>=|\bs>|@|::|\[\]|`->|`:|`:=|\^|-|\+|\bi-|\bi\+|r-|r\+|/|\*|\bdiv\b|\bi\*|\bmod\b|\br\*|~|\bi~|\br~)"
+    escape_re=r"\(({}|{})\)".format(constant_re,symbol_re)
+    const_sym_re = r"({}|{}|{})".format(constant_re,symbol_re,escape_re)
+
+    tokens = {
+        'root': [ include('elpi') ],
+
+        'elpi': [
+
+            include('_elpi-comment'),
+
+            (r"(:before|:after|:if|:name)(\s*)(\")",bygroups(Keyword.Mode,Text,String.Double),'elpi-string'),
+            (r"(:index)(\s*\()",bygroups(Keyword.Mode,Text),'elpi-indexing-expr'),
+            (r"\b(external pred|pred)(\s+)({})".format(const_sym_re),bygroups(Keyword.Declaration,Text,Name.Function),'elpi-pred-item'),
+            (r"\b(external type|type)(\s+)(({}(,\s*)?)+)".format(const_sym_re),bygroups(Keyword.Declaration,Text,Name.Function),'elpi-type'),
+            (r"\b(kind)(\s+)(({}|,)+)".format(const_sym_re),bygroups(Keyword.Declaration,Text,Name.Function),'elpi-type'),
+            (r"\b(typeabbrev)(\s+)({})".format(const_sym_re),bygroups(Keyword.Declaration,Text,Name.Function),'elpi-type'),
+            (r"\b(accumulate)(\s+)(\")",bygroups(Keyword.Declaration,Text,String.Double),'elpi-string'),
+            (r"\b(accumulate|namespace|local)(\s+)({})".format(constant_re),bygroups(Keyword.Declaration,Text,Text)),
+            (r"\b(shorten)(\s+)({}\.)".format(constant_re),bygroups(Keyword.Declaration,Text,Text)),
+            (r"\b(pi|sigma)(\s+)([a-zA-Z][A-Za-z0-9_ ]*)(\\)",bygroups(Keyword.Declaration,Text,Name.Variable,Text)),
+            (r"\b(constraint)(\s+)(({}(\s+)?)+)".format(const_sym_re),bygroups(Keyword.Declaration,Text,Name.Function),'elpi-chr-rule-start'),
```

Comment:
`\s+` should be matched as `Text.Whitespace`, not `Text`.

Reply:
done